### PR TITLE
fix: [CDS-34390]: added switch option for deployment graph view on service dashboard

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
##### Summary:

Added graph view toggle for service dashboard and re-design the deployment widget

##### Jira Links:

https://harness.atlassian.net/browse/CDS-34390

##### Screenshots:

Screenshot 2022-03-07 at 10 28 11 AM

https://user-images.githubusercontent.com/98325173/156970822-579c4854-b5b1-4bc7-b21f-39613c135f4e.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Sagar-Rajput7** [comment]
retrigger titlecheck

**Comment by Sagar-Rajput7** [comment]
trigger cypress

File: src/modules/75-cd/components/Services/DeploymentsWidget/DeploymentsWidget.module.scss [hunk 1]
```diff
@@ -33,3 +33,28 @@
     border-bottom: 1px solid var(--grey-200);
   }
 }
+
+.toggleBtns {
+  align-items: center;
+  align-self: flex-start;
+  position: absolute;
+  right: 0%;
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/Services/DeploymentsWidget/DeploymentsWidget.module.scss:41]: right: 0 should work fine

File: src/modules/75-cd/components/Services/DeploymentsWidget/DeploymentsWidget.module.scss [hunk 2]
```diff
@@ -33,3 +33,28 @@
...
+
+  .chartIcon {
+    text-align: center;
+    cursor: pointer;
+    background: var(--grey-100) !important;
+    width: var(--spacing-8);
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/Services/DeploymentsWidget/DeploymentsWidget.module.scss:47]: Isn't background and other css not working without !important

File: src/modules/75-cd/components/Services/DeploymentsWidget/DeploymentsWidget.tsx [hunk 3]
```diff
@@ -200,7 +204,7 @@ export const DeploymentsWidget: React.FC<DeploymentWidgetProps> = props => {
           {
             name: getString('success'),
             data: success,
-            color: '#3dc7f6'
+            color: '#5fb34e'
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/Services/DeploymentsWidget/DeploymentsWidget.tsx:207]: I think we have defined color for success. check in uicore

**Review by bhavya-sinha — APPROVED** [review]

**Comment by chetanvekariya-harness** [comment]
This component (src/modules/10-common/components/OverviewChartsWithToggle/OverviewChartsWithToggle.tsx) is common component for this kind of graphs. 
I think if we put small effort this can be re-used here. Also, UI is looking very bad for these charts. So, that issue will be solved as well.
Could you check if we can use that common component here ?

**Comment by Sagar-Rajput7** [comment]
> Details
@chetanvekariya-harness, I tried using that component earlier - then I was having issues showing the other components along and aligned with this, so I thought of creating the buttons instead and toggling the already existing graphs with that.

I will try using that component again.
Thanks

**Comment by Sagar-Rajput7** [comment]
> This component (src/modules/10-common/components/OverviewChartsWithToggle/OverviewChartsWithToggle.tsx) is common component for this kind of graphs. I think if we put small effort this can be re-used here. Also, UI is looking very bad for these charts. So, that issue will be solved as well. Could you check if we can use that common component here ?

@chetanvekariya-harness, I tried using that component earlier - then I was having issues showing the other components (failure and frequency rates ) along and aligned with this, so I thought of creating the buttons instead and toggling the already existing graphs with that.

I will try using that component again.
Thanks

**Comment by Sagar-Rajput7** [comment]
trigger rebase

**Comment by Sagar-Rajput7** [comment]
retrigger jest

**Comment by Sagar-Rajput7** [comment]
retrigger jest

**Comment by Sagar-Rajput7** [comment]
trigger cypress

**Comment by Sagar-Rajput7** [comment]
trigger cypress

**Comment by Sagar-Rajput7** [comment]
trigger cypress

**Pull request closed without merging**